### PR TITLE
fix(api): include event in get_autocmds

### LINKED ISSUE #17549: Add `group` and `event` to nvim_get_autocmds
### Feature already in Vim?

No

### Feature description

Following up on #14661 after brief conversation on matrix, it does feel like we're missing `event` and/or `group`(as a string) depending on the parameters given to `nvim_get_autocmds`. For example, if one requests all autocmds for a plugin/group, it is not clear when those autocmds will trigger:
image
image

The same is true if you ask only the `event`:
image
image

Note that for non-lua commands it becomes obvious (accidentaly) because of the references to their own plugins.

The idea then would be to make the return value more complete so it could be presented to the end-user as a complete chunk of information regardless of the parameter given to `nvim_get_autocmds`.

### PR DESCRIPTION
**Pull request by clason**
closes #17549 

@tjdevries

### CODE REVIEW

File: src/nvim/api/autocmd.c [hunk 1]
```diff
@@ -197,6 +197,8 @@ Array nvim_get_autocmds(Dict(get_autocmds) *opts, Error *err)
             "pattern",
             STRING_OBJ(cstr_to_string((char *)ap->pat)));
 
+        PUT(autocmd_info, "event", INTEGER_OBJ(event));
```
Comment (bfredl) [inline src/nvim/api/autocmd.c:200]: any reason we cannot use `event_nr2name` ? (afaik, event id is not something we expose, so the number does not mean lot to plugins). It was mentioned in chat but I didn't get the conclusion :)

Comment (clason) [reply src/nvim/api/autocmd.c:200]: I would like to know that reason myself ;)

I just get an error about implicit declaration when I try to use it.

Comment (gpanders) [reply src/nvim/api/autocmd.c:200]: > I just get an error about implicit declaration when I try to use it.

It is declared `static` in `nvim/autocmd.c`, if you remove the `static` keyword it should "just work".

**Comment by tjdevries** [comment]
Add test, then OK from me

**Comment by clason** [comment]
@tjdevries like so?

**Comment by tjdevries** [comment]
LGTM (didn't check why some builds were failing)

**Comment by clason** [comment]
Standard Github flake, I'm afraid. It's getting worse.

**Pull request merged**